tests/integration/connection_pool_integration.cpp

```diff
@@ -0,0 +1,150 @@
+#include <ozo/connection_info.h>
+#include <ozo/connection_pool.h>
+#include <ozo/query_builder.h>
+#include <ozo/request.h>
+#include <ozo/shortcuts.h>
+
+#include <boost/asio/spawn.hpp>
+
+#include <algorithm>
+#include <future>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+namespace {
+
+namespace asio = boost::asio;
+
+using namespace testing;
+
+TEST(connection_pool, get_connection_twice_should_get_the_same) {
```

Comment:
It is better to choose  `connection_pool_integration` name to avoid conflict with other `connection_pool` tests& Since we link all the tests in a single binary we, unfortunately, have such a problem.

Reply:
Done